### PR TITLE
backend: (csl) Add csl dialect

### PR DESCRIPTION
**Pull request by AntonLydike**
Adds the first few ops for the CSL dialect, currently only a function op, module imports and struct accesses.

### CODE REVIEW

File: xdsl/dialects/csl.py [hunk 1]
```diff
@@ -0,0 +1,250 @@
+from __future__ import annotations
```
Comment (superlopuh) [inline xdsl/dialects/csl.py:1]: Can you please add a docstring here explaining what this is and who it's for?

File: xdsl/dialects/csl.py [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+
+    def print(self, printer: Printer):
+        print_return_op_like(printer, self.attributes, self.operands)
+
+    @classmethod
+    def parse(cls, parser: Parser) -> ReturnOp:
+        attrs, args = parse_return_op_like(parser)
+        op = ReturnOp(*args)
+        op.attributes.update(attrs)
+        return op
```
Comment (PapyChacal) [inline xdsl/dialects/csl.py:235]: Sidenote: this gave me the push to do a bugfix #2475 and assuming this fix, this can use declarative syntax instead.

**Comment by superlopuh** [comment]
What's the latest on this? Are you waiting for reviews? If so, could you please split this PR into reviewable chunks?

**Comment by AntonLydike** [comment]
Closed in favor of #2591, #2593

**Pull request closed without merging**